nemo/collections/nlp/data/language_modeling/megatron/dataset_utils.py

```diff
@@ -539,6 +539,264 @@ def make_text_memmap_bin_compatibility(text_memmap_ds):
     return text_memmap_ds
 
 
+def get_dataset(
+    indexed_dataset,
+    start_index,
+    end_index,
+    cfg,
+    trainer,
+    num_samples,
+    masked_lm_prob,
+    short_seq_prob,
+    binary_head,
+    max_seq_length_dec,
+    dataset_type='standard_bert',
+    tokenizer=None,
+    max_ngram_size=3,
+    mean_ngram_size=None,
+    geometric_dist=True,
+    permutation=False,
+    whole_word_masking=True,
+    favor_long_ngrams=False,
+    delete_mask_prob=0,  # This flag is used in BART only, and will not have effect on T5/BERT
+    respect_document_boundaries=True,
+    **kwargs,
+):
+
+    if dataset_type not in DSET_TYPES:
+        raise ValueError("Invalid dataset_type: ", dataset_type)
+
+    # from nemo.collections.nlp.data.language_modeling.megatron.ict_dataset import ICTDataset
+    from nemo.collections.nlp.data.language_modeling.megatron.bert_dataset import BertDataset
+    from nemo.collections.nlp.data.language_modeling.megatron.t5_dataset import T5Dataset
+    from nemo.collections.nlp.data.language_modeling.megatron.ul2_dataset import UL2Dataset
+    from nemo.collections.nlp.data.language_modeling.megatron.bart_dataset import BARTDataset
+    from nemo.collections.nlp.data.language_modeling.megatron.length_distribution_type import LengthDistribution
```

Comment:
Why are these imports here and not at the top of the file?

Reply:
These were here previously as well and it's the same in Megatron-LM, I think it is to avoid circular imports. It shows up as new changes because of the refactor.